x-pack/platform/plugins/private/monitoring/public/plugin.ts

```diff
@@ -98,13 +101,19 @@ export class MonitoringPlugin
       mount: async (params: AppMountParameters) => {
         const [coreStart, pluginsStart] = await core.getStartServices();
         const externalConfig = this.getExternalConfig();
+        // Check if user has enterprise license
+        const license = await licensing.license$.pipe(take(1)).toPromise();
+        const hasEnterpriseLicense = license?.hasAtLeast('enterprise') || false;
```

Comment:
@ori-shafir @arnonst Unsure if we want to go through the effort (perhaps as a bug fix later), but do we want to expose this banner for trial users too?